pipeline/plot/behavior_plot.py

```diff
@@ -346,3 +348,99 @@ def compute_insta_phase_amp(data, fs, freq_band=(5, 15)):
         return insta_amp.reshape((trial_count, time_count)), insta_phase.reshape((trial_count, time_count))
     else:
         return insta_amp, insta_phase
+
+
+def get_event_locked_tracking_insta_phase(trials, event, d_name):
+    """
+    Get instantaneous phase of the jaw movement, at the time of the specified "event", for each of the specified "trials"
+    :param trials: query of the SessionTrial - note: the subsequent fetch() will be order_by='trial'
+    :param event: "event" can be
+     + a list of time equal length to the trials - specifying the time of each trial to extract the insta-phase
+     + a single string, representing the event-name to extract the time for each trial
+        In such case, the "event" can be the events in
+            + experiment.TrialEvent
+            + experiment.ActionEvent
+        In the case that multiple of such "event_name" are found in a trial, the 1st one will be selected
+            (e.g. multiple "lick left", then the 1st "lick left" is selected)
+    :param d_name: any attribute name under the tracking.Tracking table - e.g. 'jaw_y', 'tongue_x', etc.
+    :return: list of instantaneous phase, in the same order of specified "trials"
+    """
+
+    trials = trials.proj()
+
+    tracking_fs = (tracking.TrackingDevice & tracking.Tracking & trials).fetch('sampling_rate')
```

Comment:
can just do .fetch1 here

Reply:
I want to make sure we catch the cases where users specify trials from different recording type (hence different fs)

Reply:
ahh.. missed the 'set' part. :)